### PR TITLE
feat(api): expose transaction kind in transactions API

### PR DESCRIPTION
**Pull request by Radoslaw-Dabrowski**
The transactions API returns `transfer` only when a transaction has a paired counterparty transaction. A one-sided transfer-like transaction (e.g. `kind=funds_movement` moving money to an account outside this instance) has no pair, so it's indistinguishable from a regular income/expense in the API response.

This adds `transaction.kind` (`standard`, `funds_movement`, `cc_payment`, `loan_payment`, `one_time`, `investment_contribution`) to the transaction serializer, next to the existing `classification` field, so API consumers can recognize these cases too.

## Test plan
- Manually verified the field renders correctly for transactions with and without a paired transfer.

## Summary by CodeRabbit

- **New Features**
  - Added a `kind` field to v1 transaction API responses, allowing consumers to identify the transaction type even when no paired transfer is available.

### CODE REVIEW

**Review by sure-design — CHANGES_REQUESTED** [review]
The serializer change looks correct, but per AGENTS.md, any addition to `app/controllers/api/v1/` endpoints requires two things before merge:

1. **Minitest behavioral coverage** — a test in `test/controllers/api/v1/` exercising the new `kind` field.
2. **Documentation-only rswag spec** — a new spec in `spec/requests/api/v1/` (no behavioral assertions, matching the `X-Api-Key` authentication pattern), plus a regenerated `docs/api/openapi.yaml` via `RAILS_ENV=test bundle exec rake rswag:specs:swaggerize`.

See the [API checklist](docs/llm-guides/api-endpoint-consistency.md). The functional change is a one-liner; the coverage is the blocker.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/views/api/v1/transactions/_transaction.json.jbuilder [hunk 1]
```diff
@@ -22,6 +22,13 @@ json.source transaction.entry.source
 json.user_modified transaction.entry.user_modified
 json.classification transaction.entry.classification
 
+# Transaction kind (standard, funds_movement, cc_payment, loan_payment, one_time,
+# investment_contribution). Lets API consumers recognize transfer-like transactions
+# (e.g. funds_movement) even when there's no paired counterparty transaction, which
+# `transfer` alone can't express — a transfer to an account outside this instance is
+# one-sided and `transfer` stays nil for it.
```
Comment (jjmata) [inline app/views/api/v1/transactions/_transaction.json.jbuilder:29]: Missing required Minitest + rswag coverage for the new `kind` field, per AGENTS.md's API-changes rule ("Adding or modifying `app/controllers/api/v1/` endpoints requires Minitest behavioral coverage and corresponding documentation-only rswag specs... Reusable schemas belong in `spec/swagger_helper.rb`... regenerate `docs/api/openapi.yaml`").

Concretely:
- `test/controllers/api/v1/transactions_controller_test.rb`'s "transaction JSON should have expected structure" test lists every top-level key the endpoint returns but wasn't updated to assert `transaction_data.key?("kind")`, and there's no new test covering the PR's own motivating scenario (a one-sided `funds_movement` transaction where `transfer` is nil but `kind` should surface it).
- The shared `Transaction` schema in `spec/swagger_helper.rb` (~line 861-885) and `docs/api/openapi.yaml` were not updated to include `kind`, even though the sibling `TransferTransactionSide` schema already documents `kind: { type: :string }` for the same field on `_transaction_side.json.jbuilder`. Consumers reading the generated OpenAPI docs won't see this new field at all.

Suggest adding a controller test asserting the key and value for a `funds_movement`-kind transaction, adding `kind` to the `Transaction` schema in `spec/swagger_helper.rb` (mirroring `TransferTransactionSide`), and regenerating `docs/api/openapi.yaml` via `RAILS_ENV=test bundle exec rake rswag:specs:swaggerize`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by jjmata — COMMENTED** [review]
Single-pass code review (medium effort, Agent tool unavailable in this context — no multi-agent fan-out or subagent verify pass ran). One finding posted inline; diff is otherwise small and correct.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by jjmata — CHANGES_REQUESTED** [review]
Can you address the PR review comments, @Radoslaw-Dabrowski? 🙏

**Review by thewillwilson — CHANGES_REQUESTED** [review]
Exposing `kind` makes sense, and `transaction.kind` maps cleanly to the enum in `app/models/transaction.rb:70`. There have been no new commits since the earlier reviews, though, so the API-change requirements from AGENTS.md are still open:

- still open: Minitest coverage. Please add `kind` to the "transaction JSON should have expected structure" test (`test/controllers/api/v1/transactions_controller_test.rb:831`). Please also add a test for the case that motivated this PR: a one-sided `funds_movement` transaction where `transfer` is nil and `kind == "funds_movement"`.
- still open: docs. Please add `kind` (string, with the enum values) to the reusable `Transaction` schema in `spec/swagger_helper.rb` (~line 883), then regenerate `docs/api/openapi.yaml` with `RAILS_ENV=test bundle exec rake rswag:specs:swaggerize`.

Minor: the 5-line comment in the jbuilder could shrink to one line. The rationale belongs in the commit message.
